src/DependencyLoader.php

```diff
@@ -19,23 +21,36 @@ public function __construct(DependencyParser $dependencyParser)
         $this->dependencyParser = $dependencyParser;
     }
 
+    /**
+     * @throws CommandExecutionException
+     */
     public function loadDependencies(string $composer, string $project): array
     {
         $commandOutput = $this->runComposerLicenseCommand($composer, $project);
 
-        return $this->dependencyParser->parse(join(PHP_EOL, $commandOutput));
+        return $this->dependencyParser->parse(implode("\n", $commandOutput));
     }
 
-    private function runComposerLicenseCommand(string $composer, string $project, string $format = 'json'): array
+    /**
+     * @throws CommandExecutionException
+     */
+    private function runComposerLicenseCommand(string $composer, string $project): array
     {
-        $command = sprintf('%s licenses -f %s -d %s', $composer, $format, $project);
+        $command = sprintf('%s licenses --format json --working-dir %s', escapeshellarg($composer), escapeshellarg($project));
 
         return $this->exec($command);
     }
 
-    protected function exec(string $command)
+    /**
+     * @throws CommandExecutionException
+     */
+    protected function exec(string $command): array
     {
-        exec($command, $output);
+        exec($command, $output, $exitCode);
+
+        if ($exitCode !== 0) {
+            throw new CommandExecutionException('Error when trying to fetch licenses from Composer', Command::INVALID);
```

Comment:
What do you think of passing `$exitCode` through instead of using `Command::INVALID`?

Reply:
This tool should have it's own set of exit codes. I figured it is a different bad thing if a prerequisite of license checking goes wrong, i.e. the call of "composer license", and that this error case is different from "a license was found on the blocklist" or "a license was found not on the allowlist".

So currently it is:
0: Nothing wrong, no offending licenses found
1: License issue found
2: Internal technical error - in this case, a problem calling the inner composer command.

Returning an arbitrary non-zero exit code makes it impossible to be sure that 1 is always a license issue (if someone wants to check for that).

This needs an entry in the documentation.

Reply:
Added a short paragraph in README for this.